### ISSUE TITLE
Invalid shared secret length

### ISSUE THREAD

**Issue by davama** [JeNeSuisPasDave/authenticator#1]
Hello,

Really like the app. Trying to see if this works for scripting purposes.
Unfortunately when i run `add` i get:

```
authenticator add Google:dvmacias@test-server1
Enter passphrase: 
Enter shared secret: <supperlongsecretwith26chars>
Invalid shared secret string. Wrong length, incorrect padding, or embedded whitespace.
```
Not sure why that is. Any input?
Using [google-auth](https://github.com/google/google-authenticator-libpam)

thanks
-dave

**Comment by JeNeSuisPasDave** [comment]
@davama you can script it if you have `expect` available on your system (OSX via MacPorts: `sudo port install expect`; Ubuntu: `apt-get install expect`, etc.).

`expect` is generally how interactive prompt issues in scripts are handled in a Unix/Linux environment.

Here is an expect script that adds the account from your example:

```tcl
#! /usr/bin/expect
spawn authenticator add Google:dvmacias@test-server1
expect " passphrase: "
send "My secret passphrase\r"
expect " secret: "
send "HQYI3WJUKZJDKVAPPLG56IIOAAWHNKVR\r"
interact
```
I tested the above example on my system and it works.

You could get fancy and put your passphrase in a private file that the script reads and uses. Probably better than hardcoing the passphrase in the script. Googling TCL or expect should help you figure that out, should you be interested. Maybe post your solution here so others can benefit.

Note: glad you find the utility useful. If you find a better way to handle this than `expect`, please let me know.

**Comment by davama** [comment]
Thanks for replying @JeNeSuisPasDave

My issue is not the scripting but that the google auth gives me a secret longer than 16char which `authenticator` does not recongnize. If i use Google Auth App on the phone with that long secret it's no problem. 
Sorry if i wasnt clear

**Comment by JeNeSuisPasDave** [comment]
Google gives me 32-character secrets ... the example script above supplies a secret string very much like that that Google generates.

Note that 31 or 33 characters would fail ... there are only certain lengths that work.

Also the README for this project shows an example for GMail two-factor auth that has a 32 character secret ... it shows entering the code with embedded spaces, just like Google presents the code in the UI.

Can you capture a terminal session that shows the error (you can change some of the secret characters to protect your account)?

**Comment by davama** [comment]
Unfortunately i dont get 32 chars but 26 with the [this](https://github.com/google/google-authenticator-libpam#google-authenticator-pam-module). not sure why. 

```
 authenticator add dvmacias@datacom-test-server1
Enter passphrase: 
Enter shared secret: T4XV HYDW 3ZBE FDVT Y526 HSWB 7V
Invalid shared secret string. Wrong length, incorrect padding, or embedded whitespace.
Enter shared secret: T4XVHYDW3ZBEFDVTY526HSWB7V
Invalid shared secret string. Wrong length, incorrect padding, or embedded whitespace.
Enter shared secret: BZGCTPDB6HO5IZSM
OK
```
I tested with other tools and it works fine

```
oathtool --totp -b BZGCTPDB6HO5IZSM
122423
oathtool --totp -b T4XVHYDW3ZBEFDVTY526HSWB7V
622941
```

I guess i could use the other tool. But dont know why it doesnt work here. 
Thanks!

**Comment by JeNeSuisPasDave** [comment]
OK, thanks. Sorry for the misunderstandings.

I'll debug this, probably this weekend.

**Comment by JeNeSuisPasDave** [comment]
Fixed in version 1.1.3. You should be able to do a `pip install --upgrade authenticator` to get the new version.

https://github.com/JeNeSuisPasDave/authenticator/commit/3caa2edebff01be3e6fd82c72ac3a0eb415cbe66

**Comment by davama** [comment]
Thanks! Works as expected
